Algorithms/Search/exponential_search.cs

```diff
@@ -0,0 +1,65 @@
+
+using System; 
+class Gfg { 
+  
+// Returns position of first 
+// occurrence of x in array 
+static int ExponentialSearch(int []arr,  
+                         int n, int x) 
+{ 
+      
+    // If x is present at  
+    // first location itself 
+    if (arr[0] == x) 
+        return 0; 
+  
+    // Find range for binary search  
+    // by repeated doubling 
+    int i = 1; 
+    while (i < n && arr[i] <= x) 
+        i = i * 2; 
+  
+
+    return BinarySearch(arr, i/2,  
+                       Math.Min(i, n), x); 
+} 
+  
+
+static int BinarySearch(int []arr, int l, 
+                        int r, int x) 
+{ 
+    if (r >= l) 
+    { 
+        int mid = l + (r - l) / 2; 
+  
+        // If the element is present 
+        // at the middle itself 
+        if (arr[mid] == x) 
+            return mid; 
+  
+        // If element is smaller than 
+        // mid, then it can only be  
+        // present n left subarray 
+        if (arr[mid] > x) 
+            return BinarySearch(arr, l, mid - 1, x); 
+  
+        // Else the element can only  
+        // be present in right subarray 
+        return BinarySearch(arr, mid + 1, r, x); 
+    } 
+  
+    // We reach here when element 
+    // is not present in array 
+    return -1; 
+} 
+  
+// Driver code 
+public static void Main() 
+{ 
+    int []arr = {2, 3, 4, 10, 40}; 
+    int n = arr.Length; 
+    int x = 10; 
+    int result = ExponentialSearch(arr, n, x); 
```

Comment:
Codacy found an issue: [Remove the unused local variable 'result'.](https://app.codacy.com/manual/siriak/C-Sharp/pullRequest?prid=6263329)